### PR TITLE
feat: rework toasts api and add mini mode support in toasts

### PR DESCRIPTION
**Pull request by sachin-into**

### CODE REVIEW

**Review by n0izn0iz — CHANGES_REQUESTED** [review]
we already have this kind of systems in useFeedback, please adapt it instead of having two providers and api for feedback

**Review by omniwired — COMMENTED** [review]
looks good, I agree with @n0izn0iz about useFeedback

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

File: packages/context/FeedbacksProvider.tsx [hunk 1]
```diff
@@ -49,6 +53,8 @@ const defaultValue: FeedbacksProviderValue = {
   loadingFullScreen: false,
   setLoadingFullScreen: () => {},
   wrapWithFeedback: () => async () => {},
+  miniToast: { message: "" },
+  setMiniToast: () => {},
```
[the remark this answers is not in the document]
Comment (sachin-into) [reply packages/context/FeedbacksProvider.tsx:57]: mini toast has 4 different status and 2 different variant style so I created new api without touching previous working  apis. Previous toast works as it used to and new api is for mini mode won't that be simple and easier to distinguish

image

[the remark this answers is not in the document]
Comment (n0izn0iz) [reply packages/context/FeedbacksProvider.tsx:57]: I don't think, a toast should be a toast, the mode can affect how it renders but we should have a single api

you can extend the current api

File: packages/components/toasts/NormalToast.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+export interface NormalToastProps {
+  title: string;
+  message?: string;
+  onPress?: () => void;
+  type?: "error" | "success";
+  topOffset?: number;
+}
+
+export const NormalToast: React.FC<NormalToastProps> = ({
```
Comment (n0izn0iz) [inline packages/components/toasts/NormalToast.tsx:26]: there is a regression in this component, on main it has an icon

Screenshot 2024-02-28 at 12 20 47

on this pr the `!` icon does not appear

**Review by n0izn0iz — CHANGES_REQUESTED** [review]

**Review by n0izn0iz — CHANGES_REQUESTED** [review]
regression in padding
this pr:
Screenshot 2024-02-29 at 11 51 26
main:
Screenshot 2024-02-29 at 11 53 34

**Review by n0izn0iz — APPROVED** [review]
LGTM :)

**Pull request merged**